### PR TITLE
EVG-2778 don't query all test results if test names not specified

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: model/task_history.go [hunk 1]
```diff
@@ -779,12 +781,60 @@ func testHistoryV2Results(params *TestHistoryParameters) ([]task.Task, error) {
...
+		close(taskChan)
+		wg := sync.WaitGroup{}
+		wg.Add(numQueryThreads)
+		resultChan := make(chan task.Task)
+		catcher := grip.NewSimpleCatcher()
+		for i := 0; i < numQueryThreads; i++ {
+			go func() {
+				defer wg.Done()
+				for t := range taskChan {
+					err := t.MergeNewTestResults() //nolint
```
Comment (tychoish) [inline model/task_history.go]: I think the lint error is the colon in the assignment. could also put this in the if statement

File: model/task_history.go [hunk 2]
```diff
@@ -779,12 +781,60 @@ func testHistoryV2Results(params *TestHistoryParameters) ([]task.Task, error) {
...
-	if err != nil {
-		return nil, errors.Wrap(err, "error merging test results")
+	if len(params.TestNames) > 0 {
+		testQuery := db.Query(formTestsQuery(params))
+		out, err := task.MergeTestResultsBulk(tasks, &testQuery)
+		if err != nil {
+			return nil, errors.Wrap(err, "error merging test results")
+		}
+		return out, nil
+	} else {
```
Comment (tychoish) [inline model/task_history.go]: I would either: 

- take out this else statement, or
- use var statements to declare out and err above line 783, and then have one return statement after the else closes

File: model/task_history.go (hunk 1 above)
Comment (john-m-liu) [reply model/task_history.go]: I need to redeclare this in each goroutine or each will share the same address for this variable. I renamed it to make it cleaner

File: model/task_history.go (hunk 2 above)
Comment (john-m-liu) [reply model/task_history.go]: Took out the else part

**Pull request merged**